### PR TITLE
Dm 2803

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2803
https://agile6.atlassian.net/browse/DM-2802

## Description - what does this code do?
1. Redesigns homepage
2. Adds ability to configure `featured` practice body text and image in admin panel.

## Testing done - how did you test it/steps on how can another person can test it 
_**HOMEPAGE REDESIGN**_
1. Run rails `db:migrate`.
1. Visit `/`.
2. If you want to see images instead of alt text for the banner image and the `featured topic` image, add code to accomplish this (ex: Practice.find_by(slug: 'some-slug').main_display_image_s3_presigned_url(:thumb).
3. Make sure the styling reflects the Figma mockups (you should not see the `featured practice` section at this time).
4. Use a browser inspection too and ensure the mobile/tablet designs match the mockups (per design, images should remain left aligned and not increase in width. All other content should be 12 columns).
5. Click on the `Learn more` link in the banner and make sure you're taken to the new `About page` (`/about`).
6. Add a search term to the search input and ensure you're taken to the search page with results (or lack thereof) that match your input.
7. Click on the `Browse all practices` link and ensure you're taken to the search page with no query or filters.
8. Click on the `Start nomination` link and ensure you're directed to the `Nominate a practice` page.

_**FEATURED PRACTICE CONFIGURATION IN ACTIVE ADMIN**_
1. Log in as an admin and visit `/admin/practices`.
2. If you currently have a `highlighted` practice, find it and click `Unfeature`.
3. Now either click `Feature` on that same practice, or choose another practice and click `Feature`.
4. Whichever practice you have featured, click on the `Edit` link for that same practice.
5. Ensure you see the `Featured practice body` and `Featured practice attachment` fields.
6. In order to prevent gaps in content, these two fields are required.
7. To test this, click `Update practice` without filling either of them in and ensure you get an error message explaining which fields were not completed.
8. Complete one field or the other, and make sure you now see an error that states only one of the fields is now not completed.
9. Complete both fields and click `Update practice`. The practice should've updated successfully and you should now see the image on the `View` page.
10. Visit `/` and make sure you now see the `featured practice` section.
11. Click on `View practice` and make sure it takes you to the show page of the `featured practice`.
12. Go back to `/admin/practices` and unfeature the featured practice.
13. Click on the `Edit` link for that same practice and make sure you don't see the `featured` fields.

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/xjaSwxytWsGGGCcc9f9Yfa/Handoff?node-id=103%3A363

## Acceptance criteria
_**DM-2803**_
- Create new banner according to new branding refresh
- Create new header with text and new title
- Add search bar to homepage
- Add “Browse all practices” which is a link to Search Page with a no queries 
- Change alignment of Covid section  
- Refresh Nominate a Practice section
- Create Subscribe section
- Alex to provide link and figure out how subscription process for Gov Delivery works

_**DM-2802**_
- Update spacing/text/anything else for Highlighted by VA section
- Make Highlighted by VA picture configurable (i.e. user can upload picture in Admin panel)
- Picture and body text is required

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/assets/stylesheets/dm/components/_links.scss [hunk 1]
```diff
@@ -29,4 +32,45 @@
       text-decoration: underline;
     }
   }
+}
+
+.dm-primary-link {
+  color: color($theme-link-color) !important;
```
Comment (emyl3) [inline app/assets/stylesheets/dm/components/_links.scss:38]: we can keep this but just fyi you can also do `@include u-text($theme-link-color, !important);` if it is not a custom color and we don't need the `important!` we can do `  @include u-text('primary-vivid');` :D

**Review by emyl3 — CHANGES_REQUESTED** [review]
I see the `app/views/users/terms_and_conditions.html.erb` is an empty file. Do we need it? Great work Josh! :D Just a couple things I noticed

File: app/views/visns/_visn_card.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
-<div class="dm-visn-card tablet:grid-col-6 desktop:grid-col-4">
+<div class="dm-visn-card tablet:grid-col-6 desktop:grid-col-4 margin-bottom-3">
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/views/visns/_visn_card.erb:1]: I removed both unused files, `terms_and_conditions` as well as `shared/_badge` 😄

[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/views/visns/_visn_card.erb:1]: I also fixed the margin for the visn cards

File: app/assets/stylesheets/dm/components/_links.scss (hunk 1 above)
Comment (JoshingYou1) [reply app/assets/stylesheets/dm/components/_links.scss]: I added the mixin here 😃

File: app/views/visns/_visn_card.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (emyl3) [reply app/views/visns/_visn_card.erb:1]: thank you!!!

File: app/assets/stylesheets/dm/components/_links.scss (hunk 1 above)
Comment (emyl3) [reply app/assets/stylesheets/dm/components/_links.scss]: I'm not seeing that change haha but this is all good regardless 😅 Was trying to share another way to change the color of the text using uswds mixins :)

**Review by emyl3 — APPROVED** [review]
Looks good to me Josh!

**Review by bradjohnson92008 — APPROVED** [review]
Looks Great Josh - everything worked for me!

**Pull request merged**